### ISSUE TITLE
Sorting in tool mode: hash-error.

### ISSUE THREAD

**Issue by u-fischer** [plk/biber#113]
I tried to setup a sorting in tool mode and used the config file from this question on tex.SX:

http://tex.stackexchange.com/questions/269242/how-to-sort-a-bib-file-with-biber

~~~~~
<?xml version="1.0" encoding="UTF-8"?>
<config>
    <sorting>
        <presort>mm</presort>
        <sort order="1">
            <sortitem order="1">presort</sortitem>
        </sort>
        <sort order="2" final="1">
            <sortitem order="1">sortkey</sortitem>
        </sort>
        <sort order="3">
            <sortitem order="1">sortname</sortitem>
            <sortitem order="2">author</sortitem>
            <sortitem order="3">editor</sortitem>
            <sortitem order="4">translator</sortitem>
            <sortitem order="5">sorttitle</sortitem>
            <sortitem order="6">title</sortitem>
        </sort>
        <sort order="4">
            <sortitem order="1">sortyear</sortitem>
            <sortitem order="2">year</sortitem>
        </sort>
        <sort order="5">
            <sortitem order="1">sorttitle</sortitem>
            <sortitem order="2">title</sortitem>
        </sort>
        <sort order="6">
            <sortitem order="1" pad_side="left" pad_width="4" pad_char="0">volume</sortitem>
            <sortitem order="2">0000</sortitem>
        </sort>
    </sorting>
</config>
~~~~~

As bib-file I used a minimal entry:

~~~~~
@misc{regular,
      author = {Paolo Italia}}
~~~~~

When I call biber with this files I get an error message:

~~~~~~
G:\Z-Test>biber --tool --configfile=anyvt-sorting.conf lit.bib
INFO - This is Biber 2.4 running in TOOL mode
INFO - Config file is 'anyvt-sorting.conf'
INFO - Logfile is 'lit.bib.blg'
INFO - Looking for bibtex format file 'lit.bib'
INFO - Decoding LaTeX character macros into UTF-8
INFO - Found BibTeX data source 'lit.bib'
Can't use an undefined value as a HASH reference at C:\Users\XXXX\AppData\Lo
cal\Temp\par-4e696c696c616e64\cache-a29446bef12fcafa74d1affb0f5c227cf9ac0385\inc
\lib/Biber/Internals.pm line 1314.
~~~~~~

and no tool-bib is created.

**Comment by plk** [comment]
Any non-trivial tool mode usage should, as per the manual, copy the default `biber-tool.conf` and just change the parts you need. That file contains the defaults for the bit which is missing here. Run `biber --tool-config` to see the location of that file.

**Comment by u-fischer** [comment]
I see, I had somehow thought it would like with a bcf load the conf by default. But it doesn't work with biber.conf either: 

I copied biber.conf to sorting.conf. 
I called `biber --tool --configfile=sorting.conf lit.bib`: This worked fine. 
I replaced in sorting.conf

~~~~~
 <sorting>
    <presort>mm</presort>
    <sort order="1">
      <sortitem order="1">citeorder</sortitem>
    </sort>
  </sorting>
~~~~~

with the `<sorting>...</sorting>` from above and got again an hash error. It looks as if biber is choking on sortitem which refer to name fields. In my  example lit.bib (with an author field) this sorting gives the hash error:

~~~~~
  <sorting>
        <presort>mm</presort>
        <sort order="1">
            <sortitem order="1">author</sortitem>
        </sort>
    </sorting>
~~~~~

**Comment by plk** [comment]
Hmm, ok, I will check this later tonight and see about always loading the default conf first.

**Comment by plk** [comment]
It does load the `biber-tool.conf` first and then overrides this with your conf file. Anyway, this was a bug now fixed in 2.5 DEV and regression test in place.
